File: backend/Domain/Models/Loja.cs
using MarketPlace.Domain.Models.DTOs;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using MarketPlace.Interfaces.IGeneric;

namespace MarketPlace.Domain.Models;

public class Loja : IResultSearch
{
    public Guid Id { get; set; }
    public string Nome { get; set; } = null!;
    public string Descricao { get; set; } = null!;
    public decimal Cnpj { get; }
    public Guid UserId { get; }
    [ForeignKey("UserId")] public ApplicationUser User { get; private set; } = null!;

    [JsonIgnore] public List<Produto>? Produtos { get; }

    public Loja()
    {
    }

    public Loja(LojaDto lojaDto, Guid userId)
    {
        Nome = lojaDto.Nome;
        Descricao = lojaDto.Descricao;
        UserId = userId;
    }
}

File: frontend/src/app/domain/models/loja.ts
import { User } from "./user";

export class Loja {
  
    id: string;
    nome: string;
    descricao: string;
    cnpj: string;
    userId: string;
    user: User;
  
    constructor(
      id: string,
      nome: string,
      descricao: string,
      cnpj: string,
      userId: string,
      user: User
    ) {
      this.id = id;
      this.nome = nome;
      this.descricao = descricao;
      this.cnpj = cnpj;
      this.userId = userId;
      this.user = user;
    }
  }
  